services/clojars/clojars-base.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const Joi = require('joi')
+const { version: versionColor } = require('../color-formatters')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService, InvalidResponse } = require('..')
+
+const clojarsSchema = Joi.object({
+  downloads: nonNegativeInteger,
+  latest_version: Joi.string().required(),
+  latest_release: Joi.string().required(),
+}).required()
+
+class BaseClojarsService extends BaseJsonService {
+  async fetch({ clojar }) {
+    const url = `https://clojars.org/api/artifacts/${clojar}`
+    return this._requestJson({
+      url,
+      schema: clojarsSchema,
+      errorMessages: {
+        404: 'not found',
+      },
+    })
+  }
+}
+
+class ClojarsVersionService extends BaseClojarsService {
```

Comment:
If this class is also intended to be a base class, then let's include `Base` somewhere in the service name. It was a little confusing at first when I saw this line `module.exports = class ClojarsVersion extends ClojarsVersionService {` in the `clojars-version.service.js` file

Reply:
renamed `ClojarsVersionService` to `BaseClojarsVersionService`